mongo/description/server.go

```diff
@@ -446,7 +423,7 @@ func (s Server) Equal(other Server) bool {
 		return false
 	}
 
-	if s.TopologyVersion != other.TopologyVersion && CompareTopologyVersion(s.TopologyVersion, other.TopologyVersion) != 0 {
+	if s.TopologyVersion != other.TopologyVersion && s.TopologyVersion.CompareToIncoming(other.TopologyVersion) != 0 {
```

Comment:
Also a bit tangential, but I think the topology version check is redundant here?

At least, in libmongoc, we only check that the topology version is not equal when comparing server descriptions:
https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/src/mongoc/mongoc-server-description.c#L1138

Reply:
I'm not sure I understand your comment, but I think we might be able to get rid of the `!=` check. The `TopologyVersion` field is a pointer and we create a new `description.Server` instance from each `isMaster` command response, so `s.TopologyVersion != other.TopologyVersion` will always be true.

The `CompareToIncoming` check seems necessary because servers shouldn't be considered equal if there's been a topology version change.

Reply:
Ah gotcha. I did not realize the != check was comparing pointers. In that case, it seems safe to remove.

Reply:
Removed.